### PR TITLE
Update process.py

### PR DESCRIPTION
**Pull request by rchurt**
added comments to help users like me who skipped the readme and came straight here

### CODE REVIEW

File: scripts/process.py [hunk 1]
```diff
@@ -114,6 +116,19 @@ def fix_canada_recovered_data(rows):
...
+        
+unpivoting_fields = [
+    { 'name': '([0-9]+\/[0-9]+\/[0-9]+)', 'keys': {'Date': r'\1'} }
+]
+
+extra_keys = [{'name': 'Date', 'type': 'string'} ]
+extra_value = {'name': 'Case', 'type': 'number'}
+
+
+## Use dataflow to call functions defiend above and clean data
```
Comment (sglavoie) [inline scripts/process.py:130]: Spelling: `defiend` → `defined`

File: scripts/process.py [hunk 2]
```diff
@@ -1,3 +1,9 @@
+'''
+This script will import and clean data from the JHU CSSE COVID-19 dataset.
+
+Dependencies: dataflows==0.0.71 (pip install dataflows)
```
Comment (sglavoie) [inline scripts/process.py:4]: The dependencies should not be listed here: this is why `requirements.txt` exists. This leads to duplicate code, additional maintenance cost (need to update twice the same information) and will likely get out of sync once we update dependencies, leading to confusion.

The process is also described here: https://github.com/datasets/covid-19#preparation

File: scripts/process.py [hunk 3]
```diff
@@ -11,21 +17,17 @@
 DEATH_US = 'time_series_covid19_deaths_US.csv'
 REFERENCE = 'https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/UID_ISO_FIPS_LookUp_Table.csv'
 
+
+## Define functions to clean data
```
Comment (sglavoie) [inline scripts/process.py:21]: This is self-explanatory. Besides, function comments should be located inside functions:

```
def my_function():
    '''My comment here.'''
```

Here, the code is simple, but generally that doesn't help in knowing to which part of the code the comment applies when left outside of code blocks. What if functions are added below and have a different purpose?

File: scripts/process.py [hunk 4]
```diff
@@ -114,6 +116,19 @@ def fix_canada_recovered_data(rows):
             row['Long'] = row.get('Long', '-106.3468')
         yield {**expected, **row}
 
+        
+## Define parameters for cleaning
```
Comment (sglavoie) [inline scripts/process.py:120]: This is reasonable, but again we do not know if this applies only for lines 122 to 124 or also for lines 126 and 127 or even until the end of the file: after all, the comment on line 21 would theoretically apply at least until here. Comments should also generally be left without a blank line below so we have visual feedback for block of code/comments that belong together.

File: scripts/process.py [hunk 5]
```diff
@@ -1,3 +1,9 @@
+'''
+This script will import and clean data from the JHU CSSE COVID-19 dataset.
```
Comment (sglavoie) [inline scripts/process.py:2]: This is true but not necessary here. The README has more information, provides a list of sources and should be read first, hence the name `README`.

**Comment by sglavoie** [comment]
**FIXED**: Closing as the PR will not be merged. The commit title and description are also lacking important information: exploring the `git log` will tell us already which file is being modified and there is no description of the change being made that can be seen in the commit description itself without having to read the code first.

**Comment by rchurt** [comment]
@sglavoie I must say, this is one of the least helpful responses I've received to a commit that I've made.

I was attempting to improve the grossly inadequate internal documentation in this script. You yourself say that "function comments should be located inside functions"—you have made no attempt to do this for any of the functions in the script. The purpose of each function is of course not documented anywhere else, so I assume that you expect users to figure out how your code works just by reading it, which I would hope you know is bad programming practice.

Instead of realizing that the purpose of my contribution was to improve the documentation and taking it upon yourself to improve it as you saw fit if you didn't agree with my method, you tried to find any excuse to avoid adding comments. So the script remains as poorly commented as it was before.

It's fine to disagree with the way a user phrased his/her contribution, but if you expect them to keep contributing, you need to work with them, not against them. I truly hope this is helpful, and that you consider reevaluating how you value and therefore treat contributors to your projects.

**Comment by sglavoie** [comment]
* Your contribution was not an example of good programming practice and integrated undesirable changes, as explained. I tried to be helpful to point out what can be improved for any future contribution you may make in any project: it's up to you to take it as plain criticism and not learn from your shortcomings, but I think I made my points clear and you'll understand why this was not merged.
* You are making many assumptions without having sufficient context. I'll leave it at that.

**Pull request closed without merging**